pkg/api/norman/customization/cluster/validator.go

```diff
@@ -416,6 +430,9 @@ func validateAKSCredentialAuth(request *types.APIContext, credential string, pre
 // validateAKSKubernetesVersion checks whether a kubernetes version is provided
 func validateAKSKubernetesVersion(spec *v32.ClusterSpec, prevCluster *v3.Cluster) error {
 	clusterVersion := spec.AKSConfig.KubernetesVersion
+	if clusterVersion == nil {
+		return nil
+	}
```

Comment:
I'm sorry, I did not mean that you should add a whole new method to validate the cluster name. What I meant was that this section here should not be here. If cluster version is nil during a create request it is invalid, it should error here. Instead of returning nil on a case-by-case basis, the call to `validateAKSKubernetesVersion` should come after a check for `if request.Method != http.MethodPost` in the main `validateAKSConfig`.

Reply:
ok, I moved it after `if request.Method != http.MethodPost` , but I want to point that for GKE and for EKS this validation check is before, should we fix also there ?